src/ucs/datastruct/bitmap.h

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_BITMAP_H_
+#define UCS_BITMAP_H_
+
+#include <ucs/sys/compiler_def.h>
+
+BEGIN_C_DECLS
+
+#define _UCS_BITMAP_BITS_IN_WORD (64)
+#define _UCS_BITMAP_WORD_INDEX(_bit_index) ((_bit_index) / _UCS_BITMAP_BITS_IN_WORD )
+#define _UCS_BITMAP_BIT_IN_WORD_INDEX(_bit_index) ((_bit_index) % _UCS_BITMAP_BITS_IN_WORD)
+
+/*
+ * Represents an n-bit bitmap, where n is a multiple of 64, by using an array
+ * of 64-bit unsigned long integers. Memory is allocated on stack
+ * 
+ * @param _bits Bits number in the bitmap
+ */
+#define _UCS_BITMAP_DECLARE_TYPE(_bits) \
```

Comment:
can be use outside I guess, better remove initial `_`

Reply:
Are you sure? I put `_` intentionally.

I don't think we'd like to see statements like `UCS_BITMAP_DECLARE_TYPE(256)` in *several* places - it might lead to conflicts. If in the future a 256-bit bitmap is needed, the correct way is to add it *only here* (unlike the `array` datastruct, for example - where each type is *unique*, here different bitmaps with same bits number are the *same* type).

Reply:
ok, makes sense to me